tools/dumppdf.py

```diff
@@ -125,16 +172,16 @@ def dumpallobjs(out, doc, mode=None):
     return
 
 
-# dumpoutline
-def dumpoutline(
-    outfp,
-    fname,
-    objids,
-    pagenos,
-    password=b"",
-    dumpall=False,
-    mode=None,
-    extractdir=None,
+# dumpoutline_legacy
+def dumpoutline_legacy(
```

Comment:
Keeping and testing the legacy version alongside your new implementation is a good call. If some part of the system stops working because it relies on the legacy version an easy, short term fix is to change the function handle to the _legacy version.